### PR TITLE
Emit SIMD flags to AOT file if SIMD actually used

### PR DESCRIPTION
**Pull request by no1wudi**

### CODE REVIEW

File: core/iwasm/interpreter/wasm_loader.c [hunk 1]
```diff
@@ -13576,6 +13581,11 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
             {
                 uint32 opcode1;
 
+#if WASM_ENABLE_WAMR_COMPILER != 0
+                /* Mark the SIMD instruction is used in this module */
+                module->is_simd_used = true;
```
Comment (wenyongh) [inline core/iwasm/interpreter/wasm_loader.c:13586]: Should we add more checks? e.g., whether a function type's param/result type is v128, whether a function's local type is v128, whether a global type is v128?

File: wamr-compiler/main.c [hunk 2]
```diff
@@ -632,6 +634,15 @@ main(int argc, char *argv[])
...
     }
 
+    /* Only if SIMD is actually used, we need to set the compiler
+     * options */
+    if (((WASMModule *)wasm_module)->is_simd_used) {
+        option.enable_simd = true;
+    }
+    else {
+        option.enable_simd = false;
+    }
```
Comment (wenyongh) [inline wamr-compiler/main.c:644]: Is it better to set the flag inside aot_create_comp_data? The `wasm_module` is passed into the function.

Comment (no1wudi) [reply wamr-compiler/main.c]: If that, handle it in aot_create_comp_context seems better.

Comment (wenyongh) [reply wamr-compiler/main.c]: Yes, maybe it is better to handle it in aot_create_comp_context.

File: core/iwasm/compilation/aot_llvm.c [hunk 3]
```diff
@@ -3068,6 +3069,17 @@ aot_create_comp_context(const AOTCompData *comp_data, aot_comp_option_t option)
...
     LLVMDisposeMessage(triple);
 
+#if WASM_ENABLE_WAMR_COMPILER != 0
+    /* Only if SIMD is actually used, we need to set the compiler
+     * options */
+    if (((WASMModule *)comp_data->wasm_module)->is_simd_used) {
+        option->enable_simd = true;
+    }
+    else {
+        option->enable_simd = false;
```
Comment (wenyongh) [inline core/iwasm/compilation/aot_llvm.c:3079]: Can just `option->enable_simd = comp_data->wasm_module->is_simd_used;`

A concern is that then `wamrc --disable-simd` is omitted, should we do more for it?

Comment (no1wudi) [reply core/iwasm/compilation/aot_llvm.c:3079]: Yes in this implementation --disable-simd is omitted, so should we raise error if --disable-simd is set but SIMD is used in the module?

Comment (wenyongh) [reply core/iwasm/compilation/aot_llvm.c:3079]: Agree, it should be better.

**Review by wenyongh — COMMENTED** [review]
LGTM

**Pull request merged**